test/core/address_utils/sockaddr_utils_fuzzer_test.cc

```diff
@@ -0,0 +1,43 @@
+//
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+#include <grpc/grpc.h>
+
+#include "src/core/lib/address_utils/sockaddr_utils.h"
+#include "src/core/lib/iomgr/resolve_address.h"
+#include "src/core/lib/uri/uri_parser.h"
+
+bool squelch = true;
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  if (size > GRPC_MAX_SOCKADDR_SIZE) return 0;
+  grpc_resolved_address address;
+  memset(&address, 0, sizeof(address));
+  memcpy(address.addr, data, size);
+  address.len = size;
+
+  absl::StatusOr<std::string> uri = grpc_sockaddr_to_uri(&address);
+  if (!uri.ok()) return 0;
+  absl::StatusOr<grpc_core::URI> parsed_uri =
+      grpc_core::URI::Parse(uri.value());
+
+  GPR_ASSERT(parsed_uri.ok());
```

Comment:
Can we go some steps further here (needn't stall this PR) and verify that:

1. `grpc_parse_uri` succeeds
2. `grpc_sock_addr_to_uri` succeeds on the output of the above
3. the two uri strings compare equal

Reply:
Craig, 

This sounds like a good idea. I tried these out locally and I see a lot of failures now. A lot of these failures I feel are majorly due to the test setup. 

Currently we arbitrary allow any data that the fuzzer creates but grpc_parse_uri performs validation when converting the URI back to resolved address. For example for IPV6 addresses the scope has to be valid which while constructing the resolved_address, the test didn't confirm it was as the grpc_resolved_address is just memcpy of the data.

I can change the logic to be more sophisticated here and only proceed with  grpc_parse_uri if the address was truly valid. If there are better ways for me to construct the resolved_address let me know.